### ISSUE TITLE
overlapstats - Test and document for 3.3.0

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#965]
---

Author Name: **Stuart Sides** (@scsides)

Original Assignee: Jai Ram Rideout

---

Test and document overlapstats for release 3.3.0

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jai Ram Rideout** (Jai Ram Rideout)
Original Date: 2011-09-26T18:59:34Z

---

It has been determined that at least one new test is needed. The test needs to verify that errors are written to an error file, if it is provided.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jai Ram Rideout** (Jai Ram Rideout)
Original Date: 2011-09-30T21:35:06Z

---

Documentation has been updated (not checked in yet) and I am waiting on Raad and Laz to give me feedback before I commit.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jai Ram Rideout** (Jai Ram Rideout)
Original Date: 2011-10-05T22:57:14Z

---

I received approval from Laz and Raad regarding the documentation, and it has been checked into trunk and merged into the release branch.

I did not add any new tests to overlapstats because there is not enough time to devise a test for the ERRORS parameter. In order to get overlapstats to fail and write errors to the ERRORS file, cube(s) with "weird" WKT polygons must be found that cause it to fail. I have talked with the processors and they are currently looking for cubes that cause overlapstats to fail, but there will not be enough time to create a test for this release since the freeze happens on Friday and I have not received any cubes that I can test with. I have made a separate Mantis post about this (issue #519).

I have also made a separate Mantis post that addresses the questions that Laz had regarding what happens when the cube list differs from the cube list used by findimageoverlaps. There will need to be significant code changes made to overlapstats to account for this, and as such will not be included in this release. It is issue #515.
